pkg/routes/router_test.go

```diff
@@ -0,0 +1,268 @@
+	url string,
+	formData map[string]string,
+) (*http.Response, error) {
+	var req *http.Request
+
+	if len(formData) == 0 {
+		// No form data
+		req = httptest.NewRequest(http.MethodPost, url, http.NoBody)
+	} else {
+		// Build and attach form data to request
+		formBody, formType, err := MultipartForm(formData)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build multipart form: %v", err)
+		}
+
+		body := bytes.NewReader(formBody.Bytes())
+		req = httptest.NewRequest(http.MethodPost, url, body)
+		req.Header.Set("Content-Type", formType)
+	}
+
+	resp, err := app.Test(req, -1)
+	if err != nil {
+		return nil, fmt.Errorf("app test request error: %v", err)
+	}
+
+	return resp, nil
+}
+
+func TestRouter(t *testing.T) {
+	app := testutil.NewTestFiberApp(t)
+	session.CreateSession()
+
+	// closure to provide app and router to testing func
+	withArgs := func(h testHandler) func(t *testing.T) {
+		return func(t *testing.T) {
+			h(app, t)
+		}
+	}
+
+	t.Log("\n--------Setup done for API testing--------\n")
+
+	t.Run(ExportCsvUrl, withArgs(VerifyExportCsvGet))
+	t.Run(ExportStubsUrl, withArgs(VerifyExportStubsGet))
+	t.Run(CreateKwdUrl, withArgs(VerifyCreateKeywordPost))
+	t.Run(CreateKwdUrl+" missing name", withArgs(VerifyCreateKeywordMissingNamePost))
+}
+
+func VerifyExportCsvGet(app *fiber.App, t *testing.T) {
+	router := NewTestRouter(t)
+	app.Get(ExportCsvUrl, Handler(router.HandleExportCsvGet))
+
+	r := httptest.NewRequest(http.MethodGet, ExportCsvUrl, http.NoBody)
+	resp, err := app.Test(r, -1)
+	if err != nil {
+		t.Fatalf("app test request error: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		t.Fatalf("unexpected status code %d", resp.StatusCode)
+	}
+}
+
+func VerifyExportStubsGet(app *fiber.App, t *testing.T) {
+	router := NewTestRouter(t)
+	app.Get(ExportStubsUrl, Handler(router.HandleExportStubsGet))
+
+	r := httptest.NewRequest(http.MethodGet, ExportStubsUrl, http.NoBody)
+	resp, err := app.Test(r, -1)
+	if err != nil {
+		t.Fatalf("app test request error: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		t.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+}
+
+func VerifyCreateKeywordPost(app *fiber.App, t *testing.T) {
+	router := NewTestRouter(t)
+	app.Post(CreateKwdUrlFull, Handler(router.HandleCreateKeywordPost))
+
+	// TODO: should we check for num of keywords before CreateKw request
+	initialKeywords, err := database.AllKeywords(router.db)
+	if err != nil {
+		t.Fatalf("error while getting keywords: %v", err)
+	}
+
+	initialKeywordsNum := 6
+	if len(initialKeywords) != initialKeywordsNum {
+		t.Fatalf(
+			"expected %d keywords but got %d",
+			initialKeywordsNum,
+			len(initialKeywords),
+		)
+	}
+
+	data := map[string]string{
+		"name": "New keyword",
+		"args": "arg1=5, arg2=10",
+		"docs": "New keyword docs.",
+	}
+
+	url := fmt.Sprintf("%s/%s", CreateKwdUrl, database.Technical)
+	resp, err := MakePostRequest(app, url, data)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if resp.StatusCode != 302 {
```

Comment:
is this check mean that we got redirected?

Reply:
update with http status codes instead. http.StatusFound indicates a successful redirect